### PR TITLE
feat: sumi rpc support read stream

### PR DESCRIPTION
**Pull request by bytemain**
### Types

- [x] 🎉 New Features

### Background or solution

previously, we just read full content of a file, if the file is so large, it will consume many memories, so our program maybe oom.

I took a look at the applicable scenarios that can be changed:
- calculate md5
- guess encoding
- load file content

### Changelog

sumi rpc support read stream

### CODE REVIEW

File: packages/connection/src/common/ws-channel.ts [hunk 1]
```diff
@@ -291,7 +274,7 @@ export const CloseProtocol = Type.object('close', {
   reason: Type.string(),
 });
 
-const serializer = oneOf([
+const serializer = oneOf7([
```
Comment (Ricbet) [inline packages/connection/src/common/ws-channel.ts:277]: 后面的7是啥

Comment (bytemain) [reply packages/connection/src/common/ws-channel.ts:277]: 入参长度为 7，因为里面的实现是用 switch 展开了 7 个 case

CleanShot 2024-03-04 at 15 43 39@2x

**Comment by bytemain** [comment]
等到 monaco 0.45 的时候，就可以接入这个 stream 了：
CleanShot 2024-03-05 at 14 23 10@2x

**Review by Ricbet — DISMISSED** [review]
LGTM

**Review by Aaaaash — APPROVED** [review]

File: packages/connection/benchmarks/frame-decoder.bench.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+/* eslint-disable no-console */
```
Comment (erha19) [inline packages/connection/benchmarks/frame-decoder.bench.ts:1]: 这个好像是多余的？

File: packages/connection/src/common/fury-extends/any.ts [hunk 3]
```diff
@@ -15,128 +13,139 @@ export enum ProtocolType {
...
     }
 
     return value;
   }
   static reviver(key: string | undefined, value: any) {
     if (value && value.$type !== undefined && value.data !== undefined) {
-      if (value.$type === 'VSCODE_URI') {
-        return Uri.parse(value.data);
+      switch (value.$type) {
+        case 'VSCODE_URI':
```
Comment (erha19) [inline packages/connection/src/common/fury-extends/any.ts:147]: VSCODE_URI 可以找个全局变量存一下

**Review by erha19 — CHANGES_REQUESTED** [review]

File: packages/connection/benchmarks/frame-decoder.bench.ts (hunk 2 above)
Comment (bytemain) [reply packages/connection/benchmarks/frame-decoder.bench.ts:1]: 嗯 写了个框架没做，我把这个文件删了吧

**Review by erha19 — APPROVED** [review]
LGTM

**Pull request merged**